### PR TITLE
(EAI-203): Refactor serve static logic + add flexible app config

### PR DESCRIPTION
**Pull request by mongodben**
Jira: https://jira.mongodb.org/browse/EAI-203

## Changes

- Refactor mongodb-chatbot-server `AppConfig` to accept `expressAppConfig`, where you can add middleware, routes, etc.
- Add `expressAppConfig` to our implementation to serve static site in non-production envs.
- Document `expressAppConfig`
- Remove non-functional `AppConfig.serveStaticSite` logic.
  - It only worked in our env building locally, not in the published package. 

## Notes

-

### CODE REVIEW

File: examples/quick-start/packages/server/src/index.ts [hunk 1]
```diff
@@ -130,7 +130,6 @@ const config: AppConfig = {
     systemPrompt,
   },
   maxRequestTimeoutMs: 30000,
-  serveStaticSite: true,
 };
```
Comment (mongodben) [inline examples/quick-start/packages/server/src/index.ts:133]: note: this wasn't being used, and shouldn't have been there.

File: packages/mongodb-chatbot-server/src/app.ts [hunk 2]
```diff
@@ -45,15 +44,16 @@ export interface AppConfig {
...
+    Can do things like add additional, routes, global middleware, etc.
+    Runs immediately after the app is instantiated,
+    before additional routes and middleware are added.
+    @example
+    (app)=> {
+      app.get("/", (req, res) => res.send({ hello: "world" }))
+    }
    */
-  serveStaticSite?: boolean;
+  additionalServerLogic?: (app: Express) => Promise<void>;
```
Comment (nlarew) [inline packages/mongodb-chatbot-server/src/app.ts:56]:
```suggestion
  appSetup?: (app: Express) => Promise<void>;
```

nit: could we rename to be a bit more specific?

**Review by nlarew — COMMENTED** [review]
The idea & execution LGTM - can we change the config property name though?

Also does this need a docs page a la "Add Custom Middleware & Routes"? Could be a separate ticket.

**Comment by mongodben** [comment]
> Also does this need a docs page a la "Add Custom Middleware & Routes"? Could be a separate ticket.

yea should have docs. think can add to that page. will include in this PR

**Review by nlarew — APPROVED** [review]

**Pull request merged**